source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -224,53 +227,6 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
     }
     context->queries_.push_back(std::move(rec));
   }
-
-  // From RFC 1035
-  // 4.1.3. Resource record format
-  //
-  // The answer, authority, and additional sections all share the same format: a variable number of
-  // resource records, where the number of records is specified in the corresponding count field in
-  // the header.
-
-  // Parse Answer Records and Additional Resource Records. This is primarily used for tests
```

Comment:
It's understandable that the DNS client request message should not contain answer records.  Just brainstorming, what's the right behavior if it does contain that? Should we simply ignore it?  Reading through RFC1035, I don't see any text to describe this situation.

Reply:
Good point. I think the RFC1035 is fuzzy and there is no clear statement for this scenario. But I found the RFC1034 [6.2.1](https://datatracker.ietf.org/doc/html/rfc1034#section-6.2.1) query example, and RFC1035 [7.3](https://datatracker.ietf.org/doc/html/rfc1035#section-7.3), stating that `- Check the header for reasonableness.  Discard datagrams which are queries when responses are expected.` for response processing. For query we can have the similar process based on the query type. Note, this is not for inverse query. The inverse query type can include the answers with empty questions. 

For our dns filter, the parsed answers are not used for our main query logic but only for testing.